translate/filters/test_checks.py

```diff
@@ -592,6 +592,10 @@ def test_printf():
     assert fails(stdchecker.printf, "I am %#100s", "Ek is %10s")
     assert fails(stdchecker.printf, "... for user %.100s on %.100s:", "... lomuntu osebenzisa i-%. I-100s e-100s:")
     assert passes(stdchecker.printf, "%dMB", "%d MG")
+    assert fails(stdchecker.printf, "Discount 10% something", "Korting tien procent")
+    assert fails(stdchecker.printf, "Discount 10%%", "Koring 10%")
+    assert fails(stdchecker.printf, "%%$", "%$")
```

Comment:
Goal of this test `"%%$", "%$"` was to check that it will fail on second string because `%` is not part of `%%` pair and not part of valid placeholder. I consider such string malformed (so does python):

```python
>>> "%d %$" % (1,)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: not enough arguments for format string
>>> "%d %$" % (1, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: unsupported format character '$' (0x24) at index 4
```

One above in same vein but malformation is caused by `%` followed by EOL:

```python
>>> "Koring 10%" % ()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: incomplete format
>>> "Koring 10%" % (1,)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: incomplete format
```